src/main/java/org/jboss/modules/xml/SystemPropertyConditionBuilder.java

```diff
@@ -0,0 +1,47 @@
+package org.jboss.modules.xml;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Supplier;
+
+class SystemPropertyConditionBuilder {
+
+    private final List<Condition> conditions = new ArrayList<>();
+
+    public boolean resolve() {
+        for(Condition condition : conditions) {
+            if(!condition.get()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public SystemPropertyConditionBuilder add(String name, String value, boolean equal) {
+        conditions.add(new Condition(name, value, equal));
+        return this;
+    }
+
+    private static class Condition {
+        private final String name;
+        private final String value;
+        private final boolean equal;
+
+        private Condition(String name, String value, boolean equal) {
+            this.name = name;
+            this.value = value;
+            this.equal = equal;
+        }
+
+        public boolean get() {
+            String pval = System.getProperty(name);
```

Comment:
This call should be privileged.

Reply:
This does not actually need to be privileged; rather, the entire call to parse the module.xml should be privileged.  At any rate we don't want to use a lambda in there.

Reply:
I have updated it to do the parsing in priviledged blocks

Reply:
What I meant to say is, the call to parse the module.xml should *already* be privileged, meaning we don't need doPrivileged at all.